docs/data/material/migration/migrating-from-deprecated-apis/migrating-from-deprecated-apis.md

```diff
@@ -0,0 +1,51 @@
+# Migrating from deprecated APIs
+
+<p class="description">Learn how to migrate away from recently deprecated APIs before they become breaking changes.</p>
+
+## Why you should migrate
+
+Features become deprecated over time as maintainers make improvements to the APIs.
+Migrating to these improved APIs results in a better developer experience, so it's in your best interest to stay up to date.
+Deprecated APIs often become breaking changes in subsequent major versions, so the sooner you migrate, the smoother the next major update will be.
+
+## Migrating
+
+Material UI provides the `deprecations/all` codemod to help you stay up to date with minimal effort.
+
+```bash
+npx @mui/codemod@latest deprecations/all <path>
+```
+
+This command runs all the current [deprecations codemods](https://github.com/mui/material-ui/tree/master/packages/mui-codemod#deprecations), automatically migrating to the updated API.
+You can run this codemod as often as necessary to keep up with the latest changes.
+
+:::info
+
+If you need to manually migrate from a deprecated API, you can find examples below for all deprecations that have been added in Material UI v5.
+If you need to run a specific codemod, those are also linked below.
+
+:::
+
+## Accordion
+
+### TransitionComponent
+
+The Accordion's `TransitionComponent` was deprecated in favor of `slots.transition` ([Codemod](https://github.com/mui/material-ui/tree/master/packages/mui-codemod#accordion-props)):
```

Comment:
I think that we should explain why we are making these changes. The very first thing as I developer I would want to understand is why am I going through this pain, what do I get? For example, https://mui.com/material-ui/migration/migration-v3/ illustrate this.

Reply:
Yes! We have two overarching deprecation initiatives: standardizing the slots pattern and removing composed classes. I considered adding an explanation of those and then linking accordingly in each deprecation. Would that make sense?